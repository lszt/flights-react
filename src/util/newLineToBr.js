import React from 'react';

const newLineToBr = str =>
  str
    ? str.split('\n').map((line, index) => <span key={index}>{line}<br/></span>)
    : str;

export default newLineToBr;
