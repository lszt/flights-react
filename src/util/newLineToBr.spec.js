import React from 'react';
import {shallow} from 'enzyme';
import newLineToBr from './newLineToBr';

describe('util', () => {
  describe('newLineToBr', () => {
    it('should return the given value if null given', () => {
      const value = null;
      const result = newLineToBr(value);
      expect(result).toBe(null);
    });

    it('should return a span with <br> for each line', () => {
      const value =
        'my\n' +
        'multi\n' +
        'line\n' +
        'text';

      const result = newLineToBr(value);

      const component = shallow(<div>{result}</div>);

      expect(component.debug()).toMatchSnapshot();
    });
  });
});
