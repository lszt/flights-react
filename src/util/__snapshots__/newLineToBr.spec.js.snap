// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util newLineToBr should return a span with <br> for each line 1`] = `
"<div>
  <span>
    my
    <br />
  </span>
  <span>
    multi
    <br />
  </span>
  <span>
    line
    <br />
  </span>
  <span>
    text
    <br />
  </span>
</div>"
`;
